yt/data_objects/data_containers.py

```diff
@@ -813,6 +813,19 @@ def create_firefly_object(
             datadir=datadir, clean_datadir=True, **kwargs
         )
 
+        ## Ensure at least one field type contains every field requested
+        kys = self.ds.derived_field_list
+        all_fields = list(zip(*kys))[1]
+        for field in fields_to_include:
+            field = field if not isinstance(field, tuple) else field[1]
+            if field not in all_fields:
+                raise YTFieldNotFound(field=field, ds=self.ds)
```

Comment:
This doesn't seem right; I think we need to check that fields passed as single keys are not ambiguous to match the rest of yt's API
This would be done like so
```suggestion
        # error if any requested field is unknown or ambiguous
        dd = self.ds.all_data()
        fields_to_include = dd._determine_fields(fields_to_include)
```

Reply:
I don't disagree with this, but I think it would be a change in behavior of the `create_firefly_object` method.  The doc for `fields_to_include` just specifies
> fields_to_include : array_like of strings
>          A list of fields that you want to include in your Firefly visualization for on-the-fly filtering and colormapping.

which to me implies a potentially ambiguous name like `"Masses"` from the test dataset would be equivalent to `("*", "Masses")` (which I don't think exists in any form), rather than a specific field.  The [4.1 doc](https://yt-project.org/doc/analyzing/fields.html#accessing-fields-without-a-field-type) implies we should just change the `fields_to_include` doc to warn about ambiguous names, but I can also see the value for allowing something like a `"Masses" ==> ("*", "Masses")` constructor for this particular method.  What is the preferred path?

Also, this particular suggestion doesn't seem to work for cases where the particle type hasn't been specified yet. For example, it's breaking the `"Temperature"` test case because while `("gas", "Temperature")` is valid, `("all", "Temperature")` isn't. But the `"gas"` type hasn't been specified yet, so it falls back to the `"all"` type here. Am I missing something from the test definition?

Reply:
> The [4.1 doc](https://yt-project.org/doc/analyzing/fields.html#accessing-fields-without-a-field-type) implies we should just change the fields_to_include doc to warn about ambiguous names

That is not correct. yt 4.0 used to raise warnings in case of ambiguity, and selected (arbitrarily) a single candidate field, but 4.1 *just errors out* in case of ambiguity, and this is the behaviour we want to enforce everywhere going forward.

I am not sure what you mean by `("*", "Masses")`, but if you mean "all fields with this name should be included, regardless of their field type", then I think that it's just equivalent to `("all", "Masses")`, and that option should already be listed in the error message that you'd get from calling `_determine_fields` as I suggested.

You are right that this is a breaking change, but I think it's required nonetheless, because it's an oversight on our part.
I don't think it'd be trivial to only deprecate the old behaviour, but if you can see a way to, that would be preferable, of course.

Reply:
> That is not correct. yt 4.0 used to raise warnings in case of ambiguity, and selected (arbitrarily) a single candidate field, but 4.1 _just errors out_ in case of ambiguity, and this is the behaviour we want to enforce everywhere going forward.

Sorry, I wasn't clear. I just meant explicitly mention in the doc that ambiguous fields weren't allowed. 
> I am not sure what you mean by `("*", "Masses")`, but if you mean "all fields with this name should be included, regardless of their field type", then I think that it's just equivalent to `("all", "Masses")`

I was thinking more along the lines of "any fields with this name should be included", rather than "all fields with this name". So e.g. if there were three particle types, and two of them had `"field1"` (like `"ptype1"` and `"ptype2"`, but not `"ptype3"`), specifying `"field1"` would bring in `("ptype1","field1")` and `("ptype2","field1")`. Since `"ptype3"` does not have `"field1"`, `("all","field1")` doesn't exist and it's not equivalent.  But really this is just a convenience, so it's not that important.

> You are right that this is a breaking change, but I think it's required nonetheless, because it's an oversight on our part. I don't think it'd be trivial to only deprecate the old behaviour, but if you can see a way to, that would be preferable, of course.

Sounds good. I also don't see an easy way to just deprecate the old behavior. I'll see if I can clarify the doc too.

Reply:
> Since "ptype3" does not have "field1", ("all","field1") doesn't exist and it's not equivalent.

IIRC, yes it is. I'm curious where could be the source of misunderstanding here. I think I checked the definition of "all" rather recently but I'll check again. (Commenting ahead of time to avoid forgetting about this)

Reply:
This comment made me realise I was mislead.
https://github.com/yt-project/yt/blob/2efdbf3beff5f82604d2689182e425f8ec29bbc4/yt/data_objects/static_output.py#L736

So indeed you are right, we don't have an idiomatic way to express "include *any* matching field".